oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java

```diff
@@ -34,6 +36,8 @@
  */
 public class CollectionUtils {
 
+    private static final int MAX_CAPACITY = 1 << 30;
```

Comment:
why do we need this?

Reply:
This is the maximum capacity of `hashmap`.